autopush/tests/test_main.py

```diff
@@ -73,13 +74,15 @@ def test_update_rotating_tables(self):
         settings.message_tables = {}
 
         # Get the deferred back
+        e = Deferred()
         d = settings.update_rotating_tables()
 
         def check_tables(result):
             eq_(len(settings.message_tables), 1)
 
         d.addCallback(check_tables)
-        return d
+        d.addBoth(lambda x: e.callback(True))
```

Comment:
In this case, it was a race issue as the test was waiting on the deferred to fire as well as check_tables. Adding another deferred in the chain remedied the test consistently.